platform/logging/LoggerAdapter.ts

```diff
@@ -0,0 +1,56 @@
+import { Logger, LogMeta } from './Logger';
+import { LogRecord } from './LogRecord';
+
+/** @internal */
+export class LoggerAdapter implements Logger {
+  constructor(private _logger: Logger) {}
+
+  /**
+   * The current logger can be updated "on the fly", e.g. when the log config
+   * has changed.
+   */
+  set logger(logger: Logger) {
```

Comment:
__question:__ what if we just introduce `setLogger/updateLogger` method instead of setter? With this we won't need getter that throws exception?

Reply:
Yeah, it's better. ✅